src/basic/log.h

```diff
@@ -197,7 +198,12 @@ void log_assert_failed_return(
                 _e < 0 ? _e : -EIO;                                     \
         })
 
-#define log_full(level, ...) (void) log_full_errno((level), 0, __VA_ARGS__)
+#define log_full(level, fmt, ...)                                      \
+        ({                                                             \
+                if (BUILD_MODE_DEVELOPER)                              \
+                        assert(!strstr(fmt, "%m"));                    \
+                (void) log_full_errno((level), 0, fmt, ##__VA_ARGS__); \
```

Comment:
This evaluates `fmt` twice. If the logging would be baked by `printf()` this would not be a problem because having a non-constant `printf()` argument results in a warning anyhow. But I am not sure whether the `log_...()` functions enforce anything similar.

Reply:
`fmt` is always a constant string for us. We set the printf attribute on it, and the compiler verifies this.